src/lib/pickles/verify.ml

```diff
@@ -148,31 +148,49 @@ let verify_heterogenous (ts : Instance.t list) =
           in
           (absorb sponge, squeeze)
         in
-        ( absorb evals.ft_eval1 ;
-          let xs = Plonk_types.Evals.to_absorption_sequence evals.evals.evals in
-          let x1, x2 = evals.evals.public_input in
-          absorb x1 ;
-          absorb x2 ;
-          List.iter xs ~f:(fun (x1, x2) ->
-              Array.iter ~f:absorb x1 ; Array.iter ~f:absorb x2 ) ) ;
+        let old_bulletproof_challenges =
+          Vector.map ~f:Ipa.Step.compute_challenges
+            statement.pass_through.old_bulletproof_challenges
+        in
+        let sg_evals1, sg_evals2 =
+          let challenge_polys =
+            let vec =
+              Vector.map
+                ~f:(fun chals ->
+                  unstage (Wrap.challenge_polynomial (Vector.to_array chals)) )
+                old_bulletproof_challenges
+            in
```

Comment:
I'm not sure I understand why the b_poly is returned as a vec

Reply:
It's the coefficients of `b_poly`.